parsl/serialize/facade.py

```diff
@@ -0,0 +1,174 @@
+from parsl.serialize.concretes import *  # noqa: F403,F401
+from parsl.serialize.base import METHODS_MAP_DATA, METHODS_MAP_CODE
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class ParslSerializer(object):
+    """ ParslSerializer gives you one interface to multiple serialization libraries
+    """
+    __singleton_instance = None
+
+    def __new__(cls, *args, **kwargs):
+        """ Make ParslSerializer a singleton using the fact that there's only one instance
+        of a class variable
+        """
+        if ParslSerializer.__singleton_instance is None:
+            ParslSerializer.__singleton_instance = object.__new__(cls, *args, **kwargs)
+        return ParslSerializer.__singleton_instance
+
+    def __init__(self):
+        """ Instantiate the appropriate classes
+        """
+        headers = list(METHODS_MAP_CODE.keys()) + list(METHODS_MAP_DATA.keys())
+        self.header_size = len(headers[0])
+
+        self.methods_for_code = {}
+        self.methods_for_data = {}
+
+        for key in METHODS_MAP_CODE:
+            self.methods_for_code[key] = METHODS_MAP_CODE[key]()
+            self.methods_for_code[key].enable_caching(maxsize=128)
+
+        for key in METHODS_MAP_DATA:
+            self.methods_for_data[key] = METHODS_MAP_DATA[key]()
+
+    def _list_methods(self):
+        return self.methods_for_code, self.methods_for_data
+
+    def pack_apply_message(self, func, args, kwargs, buffer_threshold=1e6):
+        """Serialize and pack function and parameters
+
+        Parameters
+        ----------
+
+        func: Function
+            A function to ship
+
+        args: Tuple/list of objects
+            positional parameters as a list
+
+        kwargs: Dict
+            Dict containing named parameters
+
+        buffer_threshold: Ignored
+            Limits buffer to specified size in bytes. Default is 1e6 bytes.
+        """
+        b_func = self.serialize(func, buffer_threshold=buffer_threshold)
+        b_args = self.serialize(args, buffer_threshold=buffer_threshold)
+        b_kwargs = self.serialize(kwargs, buffer_threshold=buffer_threshold)
+        packed_buffer = self.pack_buffers([b_func, b_args, b_kwargs])
+        return packed_buffer
+
+    def unpack_apply_message(self, packed_buffer, user_ns=None, copy=False):
```

Comment:
user_ns and copy are unused here - presumably leftovers from ipp compatibility attempt